### PR TITLE
Add OOMKilled metrics and VPARecommenderFail metrics

### PR DESCRIPTION
**Pull request by byjiang1996**
1. Record number of flow executions stopped due to app container OOM killed
2. Record number of vpa recommender get recommendation failure

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/metrics/ContainerizationMetricsImpl.java [hunk 1]
```diff
@@ -127,6 +129,12 @@ public void markPodAppFailure() {
   @Override
   public void markExecutionStopped() { executionStopped.mark(); }
 
+  @Override
+  public void markOOMKilled() { executionStopped.mark(); }
```
Comment (tsangz2013) [inline azkaban-common/src/main/java/azkaban/metrics/ContainerizationMetricsImpl.java:133]: These 2 are identical now, any plan to improve granularity? e.g. for OOMKilled, mark exec-stop as well as a new metrics `oomKilled.mark()`?

Comment (byjiang1996) [reply azkaban-common/src/main/java/azkaban/metrics/ContainerizationMetricsImpl.java:133]: No plan yet. There will be a huge amount of reasons why a pod moves to Execution_stopped. It is beyond the scope of this pr to improve granularity here

**Review by tsangz2013 — DISMISSED** [review]
lgtm

**Review by tsangz2013 — APPROVED** [review]
lgtm

**Pull request merged**